agent/hcp/bootstrap/bootstrap.go

```diff
@@ -443,3 +453,25 @@ func ValidateTLSCerts(cert, key string, caCerts []string) error {
 	}
 	return nil
 }
+
+// LoadManagementToken returns the management token, either by loading it from the persisted
+// token config file or by fetching it from HCP if the token file does not exist.
+func LoadManagementToken(ctx context.Context, logger hclog.Logger, client hcpclient.Client, dataDir string) (string, error) {
+	hcpCfgDir := filepath.Join(dataDir, SubDir)
+	token, err := loadManagementToken(hcpCfgDir)
+
+	if err != nil {
```

Comment:
should we log the error message as well? I could see it potentially being useful since there are a couple different errors that function can return. Might be useful to know if the token is invalid vs missing.